### PR TITLE
Dynamic Reconfigure Experiments

### PR DESCRIPTION
**Pull request by apache8080**
So I did some more research into dynamic_reconfigure. I built a basic dynamic_reconfigure server that listens to any changes made by a dynamic_reconfigure client (e.g. our web client) and then has a callback function to update the parameters.

### CODE REVIEW

File: src/mission-planner/cfg/mission_planner.cfg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+PACKAGE='mission-planner'
+import roslib; roslib.load_manifest(PACKAGE)
+
+from dynamic_reconfigure.parameter_generator import *
+
+gen = ParameterGenerator()
+
+#examples of what kind of things we could store as params.
+
+gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
```
Comment (apache8080) [inline src/mission-planner/cfg/mission_planner.cfg]: This config file has test parameters and needs to be updated with the parameters we need for the bot. The other thing we can do with dynamic reconfigure is create parameter groups so we may be able to make a parameter group for each node.

File: src/mission-planner/cfg/mission_planner.cfg [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
+gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
+gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)
+
```
Comment (apache8080) [inline src/mission-planner/cfg/mission_planner.cfg]: This config file has test parameters and needs to be updated with the parameters we need for the bot. The other thing we can do with dynamic reconfigure is create parameter groups so we may be able to make a parameter group for each node.

File: src/mission-planner/nodes/server.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
+
```
Comment (apache8080) [inline src/mission-planner/nodes/server.py]: So this server is the thing that will be listening to the dynamic_reconfigure client and making changes as needed. For now, the callback function is just logging stuff.

File: src/mission-planner/cfg/mission_planner.cfg (hunk 2 above)
Comment (apache8080) [reply src/mission-planner/cfg/mission_planner.cfg]: This config file also has to be set to executable using `chmod a+x mission_planner.cfg` and the config file also has to be added to the CMakeList.txt file which I have already done.

**Comment by abrandemuehl** [comment]
Can you update tools/ci-docker/Dockerfile to install dynamic reconfigure? Looks like the travis build doesn't have access to it

**Comment by abrandemuehl** [comment]
Also, can you .gitignore .DS_Store?

File: src/mission-planner/cfg/mission_planner.cfg (hunk 2 above)
Comment (abrandemuehl) [reply src/mission-planner/cfg/mission_planner.cfg]: Based on how you are implementing the mission planner, what do you think these variables will be? For example, if I wanted to tell the mission planner to just do the buoy task on repeat, how would I do that?

File: src/mission-planner/nodes/server.py (hunk 3 above)
Comment (abrandemuehl) [reply src/mission-planner/nodes/server.py]: Just for plannings sake, how would the mission planner react to receiving new configuration? Should it just restart everything with a new mission plan? Would it have a subset of parameters that control the mission and require restarting missions?

**Comment by abrandemuehl** [comment]
One snag I've just found with dynamic_reconfigure is that it by nature is not persistent. We'd have to figure out how to make it persistent

File: src/mission-planner/nodes/server.py (hunk 3 above)
Comment (apache8080) [reply src/mission-planner/nodes/server.py]: I think restarting everything with a new mission plan is the best way at the moment. I will try looking around to see if there are any other examples of ways people use dynamic reconfigure that we may be able to pull from.

**Comment by abrandemuehl** [comment]
I'm closing this since it doesn't look like we'll be using it.

**Pull request closed without merging**